src/error/GraphQLError.ts

```diff
@@ -29,13 +29,22 @@ export interface GraphQLErrorOptions {
   extensions?: Maybe<GraphQLErrorExtensions>;
 }
 
+const isGraphQLErrorSymbol = Symbol.for('GraphQLError');
+
+export function isGraphQLError(error: unknown): error is GraphQLError {
+  return (
+    typeof error === 'object' && error != null && isGraphQLErrorSymbol in error
+  );
+}
+
 /**
  * A GraphQLError describes an Error found during the parse, validate, or
  * execute phases of performing a GraphQL operation. In addition to a message
  * and stack trace, it also includes information about the locations in a
  * GraphQL document and/or execution result that correspond to the Error.
  */
 export class GraphQLError extends Error {
+  [isGraphQLErrorSymbol]: true = true;
```

Comment:
I suggest making it a static property otherwise it consumes memory on every instance.
Also, I suggest having a single static property with symbol `graphql@{git_hash}` as key and enum similar to what @yaacovCR has in #3616

Reply:
@yaacovCR I am not sure whether making it a static property would make sense as another issue this might solve in the future is to have GraphQLError like-objects that are not instances of the GraphQLErro class.

The following seems way too specific for classes:

```ts
maybeError.constructor?.[isGraphQLErrorSymbol] === true
```

vs

```ts
maybeError[isGraphQLErrorSymbol] === true
```


I am not sure whether a boolean per instance is that big of an issue and a memory monster. 🤔 

____

Regarding: "I suggest having a single static property with symbol graphql@{git_hash} as key and enum similar to what @yaacovCR has in https://github.com/graphql/graphql-js/pull/3616"

I am not sure whether I get this right, but do you assume that each symbol should be scoped to a git hash commit in order to avoid cross-version usage?

 In that case, I would rather propose the solution mentioned in https://github.com/graphql/graphql-js/issues/3603#issue-1246255464 and raise an error as soon as a second GraphQL.js version is loaded which IMHO should be a separate pull request.

> Instead could be another solution to add something like this to the "entry" modules to ensure only one version of GraphQL.js is loaded? Are there use-cases in which you would want to run two graphql-js versions alongside each other?
>
>```ts
>if (globalThis[`_______graphqlVersion`] && globalThis[`_______graphqlVersion`] !== version) {
>  throw new Error("Please fix your resolutions, multiple graphql-js versions are loaded at the same time. <explanation and link for setting yarn and npm resolution>")
>}
>
>globalThis[`_______graphqlVersion`] = version
>```

Reply:
I am not sure whether we need this check at all. Our runtime code should not address wrongly configured/installed dpendencies. Only installing and loading a single version of graphql-js should be a package manager concern, not ours.

Reply:
But I generally agree with @n1ru4l that we should seriously consider removing this check. It made sense for the time when the library moved from being a regular dependency to a peer dependency, but that time was long ago.